### PR TITLE
Add dark mode support.

### PR DESCRIPTION
**Pull request by Linda-Njau**
- Implement dark mode toggle functionality, allowing users to switch between light and dark modes.
-  Add state persistence by saving a user's mode preference in `localStorage`. 
- Apply CSS changes for dark mode compatibility.

### CODE REVIEW

File: src/Nav.js [hunk 1]
```diff
@@ -27,6 +28,10 @@ class Nav extends Component {
                         <HeaderMenuItem href="https://GitHub.com/riscv/sail-riscv">RISC-V Sail Model</HeaderMenuItem>
                         <HeaderMenuItem href="https://GitHub.com/ThinkOpenly/riscvdecode">RISC-V Sail to JSON</HeaderMenuItem>
                     </HeaderNavigation>
+                    <button id="theme-switch" onClick={toggleDarkMode}>
+                    <svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 -960 960 960" width="24px" fill="#5f6368"><path d="M480-120q-150 0-255-105T120-480q0-150 105-255t255-105q14 0 27.5 1t26.5 3q-41 29-65.5 75.5T444-660q0 90 63 153t153 63q55 0 101-24.5t75-65.5q2 13 3 26.5t1 27.5q0 150-105 255T480-120Z"/></svg>
+                    <svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 -960 960 960" width="24px" fill="#e8e8e8"><path d="M480-280q-83 0-141.5-58.5T280-480q0-83 58.5-141.5T480-680q83 0 141.5 58.5T680-480q0 83-58.5 141.5T480-280ZM200-440H40v-80h160v80Zm720 0H760v-80h160v80ZM440-760v-160h80v160h-80Zm0 720v-160h80v160h-80ZM256-650l-101-97 57-59 96 100-52 56Zm492 496-97-101 53-55 101 97-57 59Zm-98-550 97-101 59 57-100 96-56-52ZM154-212l101-97 55 53-97 101-59-57Z"/></svg>
```
Comment (ThinkOpenly) [inline src/Nav.js:33]: What's this?

**Comment by ThinkOpenly** [comment]
I just merged the "formats" branch into main, because those two branches were drifting apart, needlessly. This unfortunately broke this PR. Could you rebase? (Sorry!)

File: src/Nav.js (hunk 1 above)
Comment (Linda-Njau) [reply src/Nav.js:33]: Those are the sun and moon icons used as buttons to toggle between light and dark mode. SVGs, in particular, are a type of image format that's generally smaller in size and more editable compared to something like a JPEG. SVGs are preferred for their ease of styling and scalability. I know they can be a bit of an eyesore and there are options to move them to a separate file and import them with the  tag.  On the other hand, having them inline makes it easier to apply styles and is also a quite common approach in the Frontend world.

**Comment by Linda-Njau** [comment]
> I just merged the "formats" branch into main, because those two branches were drifting apart, needlessly. This unfortunately broke this PR. Could you rebase? (Sorry!)

Done : )

**Review by ThinkOpenly — APPROVED** [review]

**Pull request merged**